core/CronArchive/SegmentArchiving.php

```diff
@@ -121,11 +124,19 @@ public function getSegmentArchivesToInvalidate($idSite, $checkOnlyForNewSegments
     public function findSegmentForHash($hash, $idSite)
     {
         foreach ($this->getAllSegments() as $segment) {
-            if (!$this->isAutoArchivingEnabledFor($segment)) {
+            if (!$this->isAutoArchivingEnabledFor($segment)
+                || !$this->isSegmentForSite($segment, $idSite)
+            ) {
+                continue;
+            }
+
+            try {
+                $segmentObj = new Segment($segment['definition'], [$idSite]);
+            } catch (\Exception $ex) {
+                $this->logger->debug("Could not process segment {$segment['definition']} for site {$idSite}. Segment should not exist for the site, but does.");
```

Comment:
The debug message confuses me. Isn't an exception thrown when the segment does not exist?
Also this case shouldn't occur at all if the `isSegmentForSite` works correctly, right?

Reply:
It shouldn't occur, but if it does it would break archiving, and I didn't think that needed to happen. This is a sanity check.